endpoint/platform/wp77xx/wp77xx.c

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (C) 2019-2020 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <termios.h>
+#include <time.h>
+#include <unistd.h>
+#include "endpoint/hal/device.h"
+
+#include "legato.h"
+
+#include "interfaces.h"
+#include "le_log.h"
+#include "le_mdmDefs_interface.h"
+#include "le_secStore_common.h"
+#include "le_sim_common.h"
+
+#define MAXLINE 1024
+
+#define CBC_IV_SIZE 16
+#define IMSI_LEN 15
+#define READ_BUFFER_SIZE 32
+#define DEFAULT_PORT "/dev/ttyHS0"
+
+static inline void register_wp77xx(void);
+static inline void unregister_wp77xx(void);
+
+extern struct device_type wp77xx_device_type;
+
+static le_sim_Id_t SimId;
+/* UART file descriptor */
+static int uart_fd;
+
+static status_t wp77xx_init(void) {
+  register_wp77xx();
+  return SC_OK;
+}
+
+static void wp77xx_release(void) { unregister_wp77xx(); }
+
+static status_t wp77xx_get_key(uint8_t *key) {
+  // FIXME:Need to implement the private key generate algorithm
+  if (key == NULL) {
+    LE_ERROR("Failed to get key");
+    return SC_ENDPOINT_GET_KEY_ERROR;
+  }
+
+  static char *private_key = "AAAAAAAAAAAAAAAA";
```

Comment:
Don't set `static` here. Compilers might generate worse code for such local variables. Use godbolt.org
 to check. Instead, it should be a straightforward string literal.